tools/custom-organization-policy-library/build/custom-constraints/gke/gkeRequireNetworkPolicy.yaml

```diff
@@ -5,8 +5,7 @@
 name: #@ constraint.constraint_name()
 resource_types: 
 - container.googleapis.com/Cluster
-- container.googleapis.com/NodePool 
-condition: 1 == 0
+condition: resource.addonsConfig.networkPolicyConfig.disabled == true
```

Comment:
More complex this is one.
The contact should if:
```
if control plane v1 and resource.addonsConfig.networkPolicyConfig.disabled == true then violations
```
As by default, control plane v2 has network policy enabled

Reply:
Yes understood. This constraint is more complex, since by enforcing data plane V2 , it will also enable network policy too. I also do not see Data plane v1 in GCP, is it retired?

Reply:
By default it is using dataplane v1 i believe.

But to enable dataplane v2 here are the step by API https://cloud.google.com/kubernetes-engine/docs/how-to/dataplane-v2#api

So the conditions should be:
```
- if not dataplane v2 (as explain in link) and if resource.addonsConfig.networkPolicyConfig.disabled == true then violations
```
Something like that (not tested so i will have to adapt to the real case)
```
- (if resource.networkConfig.datapathProvider != "ADVANCED_DATAPATH" and resource.addonsConfig.networkPolicyConfig.disabled == true) then violations
```

Reply:
Alright! Thanks Vannick! I will test and try to adapt to the real case and update you if it works.

Reply:
![Screenshot 2024-04-16 10 43 45 AM](https://github.com/vannicktrinquier/professional-services/assets/162395680/d5fe2e37-61c1-4a44-9659-866f608b953e)
Hi Vannick, I have tested the dataplane v2 constraint works using the concept of your condition provided in the comment. 
Some minor changes were done to the code to ensure the constraint work in GCP.

**condition: resource.networkConfig.datapathProvider != 'ADVANCED_DATAPATH' && resource.addonsConfig.networkPolicyConfig.disabled == true**